config/config.go

```diff
@@ -584,18 +598,57 @@ type MarathonSDConfig struct {
 	XXX map[string]interface{} `yaml:",inline"`
 }
 
+// KubernetesSDConfig is the configuration for Kubernetes service discovery.
+type KubernetesSDConfig struct {
+	Server           string   `yaml:"server"`
+	NodeMetricsPort  int      `yaml:"node_metrics_port,omitempty"`
+	InCluster        bool     `yaml:"in_cluster,omitempty"`
+	BearerTokenFile  string   `yaml:"bearer_token_file,omitempty"`
+	Username         string   `yaml:"username,omitempty"`
+	Password         string   `yaml:"password,omitempty"`
+	Insecure         bool     `yaml:"insecure,omitempty"`
+	CertFile         string   `yaml:"cert_file,omitempty"`
+	KeyFile          string   `yaml:"key_file,omitempty"`
+	CAFile           string   `yaml:"ca_file,omitempty"`
+	RetryInterval    Duration `yaml:"retry_interval,omitempty"`
+	RequestTimeout   Duration `yaml:"request_timeout,omitempty"`
+	DiscoverServices bool     `yaml:"discover_services,omitempty"`
```

Comment:
@brian-brazil The reason for this is to enable/disable discovery of kubernetes services. With this enabled (default), we watch all services & check the scrape key to see whether it should be scraped or not. If users don't want to enable this & just collect cluster level metrics this can be disabled: this reduces the resources for 2 more watches, long polling connections to the kubernetes master.